### DISCUSSION TITLE
After migrating the client from 1.11 to 2.0, I started getting exceptions S3Exception 301 Moved permanently

### DISCUSSION THREAD

**Discussion by skipkolch** [aws/aws-sdk-java-v2#4116] [Q&A]
Previously, the client was used to work with S3:
`implementation("com.amazonaws:aws-java-sdk-s3:1.11.939")`

The client was configured like this:
```
private val client = AmazonS3ClientBuilder.standard()
        .withCredentials(
            AWSStaticCredentialsProvider(
                BasicAWSCredentials(
                    s3CredentialConfig.accessKey,
                    s3CredentialConfig.secretKey
                )
            )
        )
        .withEndpointConfiguration(
            AwsClientBuilder.EndpointConfiguration(
                s3CredentialConfig.serviceEndpoint,
                s3CredentialConfig.region
            )
        )
        .withPathStyleAccessEnabled(true)
        .build()
```
Then I needed an asynchronous s3 client and I decided to migrate to v2.0:
```
implementation("software.amazon.awssdk:s3:2.20.12") {
        exclude("software.amazon.awssdk", "netty-nio-client")
        exclude("software.amazon.awssdk", "apache-client")
}
```
The client was configured like this:

```
private val client =  S3AsyncClient.builder()
            .httpClientBuilder(AwsCrtAsyncHttpClient.builder())
            .credentialsProvider(credentials)
            .endpointOverride(uri)
            .region(region)
            .forcePathStyle(true)
            .build()
```

**Сredits, endpoint and region did not change.**

When viewing the contents of the bucket, I get exceptions:
`software.amazon.awssdk.services.s3.model.S3Exception: null (Service: S3, Status Code: 301, Request ID: null) java.lang.Throwable: java.util.concurrent.CompletionException: software.amazon.awssdk.services.s3.model.S3Exception: null (Service: S3, Status Code: 301, Request ID: null)`

Request log:
```
@coroutine#7 - software.amazon.awssdk.auth.signer.Aws4Signer: AWS4 Canonical Request: GET
/replay
prefix=a6c8-7b0dc3d75526-1
amz-sdk-3-id:8731f78d-xxxx-5258-8f76a3ff82ba
amz-sdk-request:attempt=1; max=4
host: xxxx.xxxx.xxxx
x-amz-content-sha256:UNSIGNED-PAYLOAD
x-amz-date:20230619T085216Z
```

**Comment by debora-ito** [comment]
Can you enable the verbose wirelogs? I think S3 sends extra information in 301 response body that the SDK is not able to add to the error message.

Instructions on how to enable verbose wirelogs can be found in our Developer Guide - https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/logging-slf4j.html#sdk-java-logging-verbose
Make sure to redact any sensitive information (like access keys).

Also, keep in mind that if you are using a third-party integration with s3 (and not s3 endpoints directly) the issue might be on the third-party side, and those are not in scope of this support channel.
